keps/sig-apps/3140-TimeZone-support-in-CronJob/README.md

```diff
@@ -0,0 +1,504 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+CronJob creates Jobs based on the schedule specified by the author, but the Time
+Zone used during the creation depens on where kube-controller-manager is running.
+This proposal aims to extend CronJob resource with the ability for a user to
+define the TimeZone when a Job should be created.
+
+## Motivation
+
+Not long after the [introduction of CronJob in kubernetes](https://github.com/kubernetes/kubernetes/pull/11980)
+a [request was raised to support setting time zones](https://github.com/kubernetes/kubernetes/issues/47202).
+The initial [response from SIG-Apps and SIG-Architecture](https://github.com/kubernetes/kubernetes/issues/47202#issuecomment-360820586)
+at the time was that introducing such functionality would require cluster operator
+to manually include TimeZone database since golang did not have one.
+Starting from [golang 1.15](https://go.dev/doc/go1.15) `time/tzdata` package can
+be embedded in the binary thus removing the requirement for external database.
+At that time the majority of the focus was towards [moving CronJob to GA](https://github.com/kubernetes/enhancements/issues/19)
+so the effort to support TimeZone was again delayed. Now that we have CronJob
+fully GA-ed it's time to satisfy the original request.
+
+### Goals
+
+- Add the field `.spec.timeZone` which allows specifying a valid TimeZone
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Add the field `.spec.timeZone` to the CronJob resource. The cronjob controller
+will take the field into account when scheduling the next Job run. In case the
+field is not specified or is empty the controller will maitain the current
+behavior which is to rely on the time zone of the kube-controller-manager
+process.
+
+### Notes/Constraints/Caveats (Optional)
+
+The current mechanism, which will still be the default behavior heavily relies
+on the time zone of the kube-controller-manager process which is hard for a
+regular user to figure out. Exposing an explicit field for setting a time zone
+will allow CronJob authors to simplify the user experience when creating CronJobs.
+
+### Risks and Mitigations
+
+- Outdated time zone in the golang might lead to wrong schedule times
+
+This problem can be mitigated with a fresh build of kube-controller-manager with
+updated golang version, but it heavily relies on the go community to keep the
+time zone database up-to-date.
+
+- Malicious user can create multiple CronJobs with different time zone which can
+actually trigger Jobs at the exact same time
```

Comment:
Can't malicious user trigger Jobs at the same time without time zone?

Reply:
It's easier to notice if all jobs are expressed w/o timezone, with timezone it's slightly harder.